cumulusci/tasks/salesforce/Deploy.py

```diff
@@ -136,6 +153,68 @@ def _process_meta_xml(self, zipf):
         zipf = zip_clean_metaxml(zipf, logger=self.logger)
         return zipf
 
+    def _process_static_resources(self, zip_src):
+        relpath = self.options.get("static_resource_path", "static-resources")
+        if not os.path.exists(relpath):
+            return zip_src
+        path = os.path.realpath(relpath)
+
+        # We need to build a new zip file so that we can replace package.xml
+        zip_dest = zipfile.ZipFile(io.BytesIO(), "w", zipfile.ZIP_DEFLATED)
+        for name in zip_src.namelist():
+            content = zip_src.read(name)
+            if name == "package.xml":
+                package_xml = content
+            else:
+                zip_dest.writestr(name, content)
+
+        # Build static resource bundles and add to package zip
+        with temporary_dir():
+            os.mkdir("staticresources")
+            bundles = []
+            for name in os.listdir(path):
+                bundle_relpath = os.path.join(relpath, name)
+                bundle_path = os.path.join(path, name)
+                if not os.path.isdir(bundle_path):
```

Comment:
Uncompressed single files are legal static resources. What would you think about just copying regular files over from `static-resources` to `staticresources`? Otherwise, a project that had both single-file and directory static resources would end up with a hybrid directory structure like this, which might be confusing.

    src/
        staticresources/
            myresource.resource  (text/json)
            myotherresource.resource (text/csv)
        static-resources/
            my-resource-dir/
                file1

Reply:
Personally the hybrid doesn't bother me -- there's a clear distinction between static resources that fit the standard metadata format and can be deployed without modification, and uncompressed bundles that need to be transformed first. I'm curious to see what Jason thinks next week.